### PR TITLE
feat: Migrate to files:node:updated

### PR DESCRIPTION
**Pull request by luka-nextcloud**
### 📝 Summary

* Resolves: #4054

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A

### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

**Comment by luka-nextcloud** [comment]
> @luka-nextcloud what's the background for this change? Does it fix anything? Is it a technical dept thing?

Sorry for not mentioning. This is a technical debt, I've added the issue on the description.

**Comment by mejo-** [comment]
Thanks for the additional context @luka-nextcloud. Do you know what the emitted event is supposed to do? I.e. is there a reproducer of how to test whether this works?

**Comment by luka-nextcloud** [comment]
@mejo-  The emitted event is supposed to notify the other apps that a file was updated. Install text app and notes app (with this PR https://github.com/nextcloud/notes/pull/1377) then open notes app to edit a note.

File: src/components/Editor.vue [hunk 1]
```diff
@@ -650,7 +652,14 @@ export default {
 		},
 
 		onSave() {
-			emit('files:file:updated', { fileid: this.fileId })
+			const user = getCurrentUser()
+			const node = new File({
+				id: this.fileId,
+				source: generateRemoteUrl(`dav/files/${user.uid}${this.relativePath}`),
+				mtime: new Date(),
+				mime: 'text/markdown',
```
Comment (juliushaertl) [inline src/components/Editor.vue:660]: The text app might also handle other mime types, so we should take the one that the viewer passes along to us

**Review by juliushaertl — CHANGES_REQUESTED** [review]
Small comment

File: src/components/Editor.vue (hunk 1 above)
Comment (juliusknorr) [reply src/components/Editor.vue:660]: @luka-nextcloud Can you check that one please?

**Review by juliusknorr — APPROVED** [review]

**Comment by juliusknorr** [comment]
Was just reported as an issue so this likely fixes https://github.com/nextcloud/text/issues/6671

**Comment by juliusknorr** [comment]
- [ ] Double check and test if this resolves https://github.com/nextcloud/text/issues/6671

File: src/components/Editor.vue [hunk 2]
```diff
@@ -667,7 +669,13 @@ export default {
 		},
 
 		onSave() {
-			emit('files:file:updated', { fileid: this.fileId })
+			const node = new File({
+				id: this.fileId,
+				source: generateRemoteUrl(`dav/files/${this.currentSession.userId}${this.relativePath}`),
+				mtime: new Date(),
+				mime: this.mime,
```
Comment (juliusknorr) [inline src/components/Editor.vue:676]: @luka-nextcloud Can you explain why we need to also emit source and mime here? I fear source could break that way if we are using text on a public share link.

Comment (luka-nextcloud) [reply src/components/Editor.vue:676]: @juliusknorr I have just checked again, source and mime are mandatory.

Comment (luka-nextcloud) [reply src/components/Editor.vue:676]: @juliusknorr according to
https://github.com/nextcloud-libraries/nextcloud-files/blob/333c1fc5f64904ec73dff627118b9872992c0d28/lib/files/nodeData.ts#L80-L82
https://github.com/nextcloud-libraries/nextcloud-files/blob/333c1fc5f64904ec73dff627118b9872992c0d28/lib/files/nodeData.ts#L107-L110

**Comment by juliusknorr** [comment]
So many cypress failures seem odd, will do another rebase.

**Comment by susnux** [comment]
> [x] Double check and test if this resolves

Can we backport to 30 and 29?

**Pull request merged**